src/OrchardCore.Modules/OrchardCore.Users/Extensions/UserTimeZoneServiceExtensions.cs

```diff
@@ -0,0 +1,48 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Modules;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.TimeZone.Services;
+
+namespace OrchardCore.Users;
+
+/// <summary>
+/// Provides an extension for <see cref="IUserTimeZoneService"/>.
+/// </summary>
+public static class UserTimeZoneServiceExtensions
+{
+    /// <summary>
+    /// Gets the time zone for currently logged-in user.
+    /// </summary>
+    /// <param name="userTimeZoneService">The <see cref="IUserTimeZoneService"/>.</param>
+    public static async ValueTask<ITimeZone> GetTimeZoneAsync(this IUserTimeZoneService userTimeZoneService)
```

Comment:
GetCurrentUserAsync(this IUserTimeZoneService userTimeZoneService, HttpContext httpContext)

Reply:
This will be annoying to pass `HttpContext` in the extension method. Moreover, you need to change the visibility for a few members of `UserTimeZoneExtensions` which makes things unmanaged. You could try this at your end to see what I mean

Reply:
Then don't add the extension. In the driver that calls this service, get the current user and pass it to the method without the extension.

Reply:
We already have a user in the driver

Reply:
okay so pass that user to the method and don't waste time with the extension